src/main/native/pairhmm/IntelPairHmm.cc

```diff
@@ -101,9 +107,19 @@ JNIEXPORT void JNICALL Java_com_intel_gkl_pairhmm_IntelPairHmm_computeLikelihood
 
   //==================================================================
   // get Java data
-  JavaData javaData;
-  std::vector<testcase> testcases = javaData.getData(env, readDataArray, haplotypeDataArray);
-  double* javaResults = javaData.getOutputArray(env, likelihoodArray);
+  JavaData javaData(env);
+
+  std::vector<testcase> testcases;
+  double* javaResults;
+
+  try {
+    testcases = javaData.getData(readDataArray, haplotypeDataArray);
+    javaResults = javaData.getOutputArray(likelihoodArray);
+  } catch (JavaException& e) {
+    env->ExceptionClear();
+    env->ThrowNew(env->FindClass(e.classPath), e.message);
```

Comment:
Add `javaData.releaseData(env);` before return

Reply:
That's not necessary because it's handled in the destructor.